frontend/packages/console-dynamic-plugin-sdk/src/extensions/storage-class-provisioner.ts

```diff
@@ -0,0 +1,45 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { CodeRef, ExtensionDeclaration } from '../types';
+
+export type ProvisionerProps = {
+  onParamChange: (id: string, paramName: string, checkbox: boolean) => void;
+  parameterKey: string;
+  parameterValue: string;
+};
+
+export enum ProvisionerType {
+  CSI = 'CSI',
+  OTHERS = 'OTHERS',
+}
+
+export type ProvisionerDetails = {
+  title: string;
+  provisioner: string;
+  allowVolumeExpansion: CodeRef<(arg) => boolean>;
+  volumeBindingMode?: string;
+  documentationLink?: CodeRef<() => string>;
+  parameters?: {
+    [paramKey: string]: {
+      name: string;
+      hintText: string;
+      value?: string;
+      values?: string[];
+      visible?: CodeRef<(arg) => boolean>;
+      required?: CodeRef<(arg) => boolean>;
+      validation?: CodeRef<(params) => string>;
+      format?: CodeRef<(arg) => string>;
+      Component?: CodeRef<React.ComponentType<ProvisionerProps>>;
+      type?: 'checkbox';
+    };
+  };
+};
+
+export type StorageClassProvisioner = ExtensionDeclaration<
+  'console.storage-class/provisioner',
+  {
+    [provisionerType in keyof typeof ProvisionerType]?: ProvisionerDetails;
```

Comment:
Can you please check if the generated JSON schema looks correct?

```
packages/console-dynamic-plugin-sdk/generated/schema/console-extensions.json
```

Not sure if this TS syntax is supported by `ts-json-schema-generator` version that we're using.

Reply:
```
   "properties": {
          "type": "object",
          "properties": {
            "CSI": {
              "$ref": "#/definitions/ProvisionerDetails"
            },
            "OTHERS": {
              "$ref": "#/definitions/ProvisionerDetails"
            }
          },
```
Generated file looks good to me

Reply:
Thanks, it seems that the object keys are expanded as expected in the JSON schema.